src/OpenIddict.Core/OpenIddictProvider.cs

```diff
@@ -12,19 +12,11 @@
 using System.Threading.Tasks;
 using AspNet.Security.OpenIdConnect.Extensions;
 using AspNet.Security.OpenIdConnect.Server;
-using Microsoft.AspNet.Identity;
-using Microsoft.AspNet.Identity.EntityFramework;
-using Microsoft.Data.Entity;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Internal;
-using OpenIddict.Models;
 
 namespace OpenIddict {
-    public class OpenIddictProvider<TContext, TUser, TRole, TKey> : OpenIdConnectServerProvider
-        where TContext : OpenIddictContext<TUser, TRole, TKey>
-        where TUser : IdentityUser<TKey>
-        where TRole : IdentityRole<TKey>
-        where TKey : IEquatable<TKey> {
+    public class OpenIddictProvider<TUser, TApplication> : OpenIdConnectServerProvider where TUser : class where TApplication : class {
```

Comment:
same suggestion as in the controller, constructor inject dependent managers instead of Provider<T>.